### PR TITLE
feat(RHINENG-20360) Make the inv_publish_hosts script configurable

### PR DESCRIPTION
**Pull request by rodrigonull**
# Overview

This PR is being created to address [RHINENG-20360](https://issues.redhat.com/browse/RHINENG-20360).

It introduces 4 new environment variables that allow to set the publications that will be created and dropped. It also allows to set the replica identity mode for the replicated tables.

This will allow us to continue the logical replication tests on stage environment and be able to run this job in production when we fell that we are ready to give it another try to update the replica identity and rollback to default if needed.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [x] Tests: validate optimal/expected output
- [x] Tests: validate exceptions and failure scenarios
- [x] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [x] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

## Summary by Sourcery

Enable dynamic configuration of PostgreSQL publications, replication slot cleanup, and replica identity settings in the inv_publish_hosts script via environment variables; refactor core logic into reusable functions; update deployment manifest; and add documentation and extensive tests.

New Features:
- Drive publication creation, deletion, and replication slot cleanup via environment variables (CREATE_PUBLICATIONS, DROP_PUBLICATIONS, DROP_REPLICATION_SLOTS).
- Support setting replica identity mode (default or index) for tables and their partitions via REPLICA_IDENTITY_MODE.
- Generate CREATE PUBLICATION SQL dynamically from a central PUBLICATION_CONFIG structure.

Enhancements:
- Refactor inv_publish_hosts.py to modularize publication and replica identity setup into reusable functions with improved logging and configuration handling.

Build:
- Expose new environment variables for publication and replica identity settings in the clowdapp deployment manifest.

Documentation:
- Add user-facing documentation describing inv_publish_hosts.py configuration via environment variables.

Tests:
- Add comprehensive unit and integration tests covering publication SQL generation, publication setup, replication slot cleanup, and replica identity management.

### CODE REVIEW

File: jobs/inv_publish_hosts.py [hunk 1]
```diff
@@ -98,22 +119,42 @@
...
+    AND tablename != :table_name
+    ORDER BY tablename
+""")
+
+GET_CURRENT_REPLICA_IDENTITY_SQL = sa_text("""
+    SELECT relreplident
+    FROM pg_class
+    WHERE oid = (:qualified_table_name)::regclass
+""")
+
```
[the remark this answers is not in the document]
Comment (rodrigonull) [reply jobs/inv_publish_hosts.py:143]: It is a job script, so I prefer to keep it in a single file.

File: jobs/inv_publish_hosts.py [hunk 2]
```diff
@@ -140,53 +181,231 @@ def _excepthook(logger, exc_type, value, traceback):
...
+        table_name = table_config["name"]
+        schema = table_config["schema"]
+        columns = ",".join(table_config["publication_columns"])
+        publication_filter = table_config.get("publication_filter", "")
+
+        if publication_filter:
+            table_spec = f"""      {schema}.{table_name} ({columns})
+        WHERE ({publication_filter})"""
+        else:
+            table_spec = f"""      {schema}.{table_name} ({columns})"""
```
Comment (kruai) [inline jobs/inv_publish_hosts.py:202]: Weird spacing, but not a big deal

**Review by kruai — APPROVED** [review]

File: unit_test.sh [hunk 3]
```diff
@@ -41,12 +41,15 @@ trap "teardown_podman" EXIT SIGINT SIGTERM
...
 
+echo "wal_level = logical" > 99-wal-level.conf
+
 DB_CONTAINER_ID=$(podman run -d \
     --name "$DB_CONTAINER_NAME" \
     --network "$NETWORK" \
     -e POSTGRESQL_USER="inventory-test" \
     -e POSTGRESQL_PASSWORD="inventory-test" \
     -e POSTGRESQL_DATABASE="inventory-test" \
+    -v ./99-wal-level.conf:/etc/postgresql/conf.d/99-wal-level.conf:Z \
```
[the remark this answers is not in the document]
Comment (rodrigonull) [reply unit_test.sh:52]: This file is created on line 44.

**Review by thearifismail — APPROVED** [review]
LGTM
